Source/RuntimeImageLoader/Private/Texture2DAnimation/AsyncGIFLoader.cpp

```diff
@@ -11,7 +12,17 @@ UAnimatedTexture2D* UAsyncGIFLoader::Init(const FString& GIFFilename)
 
 	if (Decoder)
 	{
-		Decoder->GIFDecoding(TCHAR_TO_UTF8(*GIFFilename));
+		CurrentTask = Async(
+			EAsyncExecution::Thread,
+			[GIFFilename, &Decoder]()
+			{
+				Decoder->GIFDecoding(TCHAR_TO_UTF8(*GIFFilename));
+				
+				return !Decoder->GetTextureData().IsEmpty();
+			}
+		);
+		
+		bool bResult = CurrentTask.Get();
```

Comment:
This does not seem to be correct:
1) You are executing this code in Init() function. You call Init() on game thread, then you do .Get(), which effectively blocks Game Thread! 

2) Make sure you read UE docs on `BlueprintAsyncActionBase`: https://docs.unrealengine.com/5.2/en-US/API/Runtime/Engine/Kismet/UBlueprintAsyncActionBase/ 

3) Override UBlueprintAsyncActionBase::Activate(...) to actually implement async node.